### PR TITLE
Improve readme documentation with some slight refactors

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR contains enhancements on Readme with following changes
* Ably paragraph, overview
* Add instructions for usage of the library
* Add sample code blocks that showcase API functionality
* Some API refactor as a result of realisation during writing documentation

This PR does not include SDK concepts and terminology, contribution guide and code documentation

### CODE REVIEW

**Review by AndyNicks — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,24 +1,256 @@
...
+        }
+```
+As you can see, you should invoke ```enter``` inside a coroutine scope.
+You must replace ```gamePlayer``` with your own instance of ```GamePlayer```
+
+#### Leave a game
+
+```kotlinlang
+someCoroutineScope.launch {
+            val leaveResult = MultiplayerGameApp.instance.ablyGame.leave(gamePlayer!!)
```
Comment (KacperKluka) [inline README.md]: Why here you've added the `MultiplayerGameApp.instance.` and also the `!!` at the end of the line? I'd say let's make this code example look like the rest of them :wink: 
```suggestion
            val leaveResult = ablyGame.leave(gamePlayer)
```

File: README.md [hunk 2]
```diff
@@ -1,24 +1,256 @@
...
+       }
+```
+As ```allPlayers``` is a suspending function, you must call this from a coroutine scope
+
+#### Check if a player is in a game
+```
+coroutineScope.launch {
+            if (ablyGame.isInGame(gamePlayer)) {
+               //do something
+            }
```
Comment (KacperKluka) [inline README.md:149]: A closing `}` is missing :wink: 
```suggestion
coroutineScope.launch {
            if (ablyGame.isInGame(gamePlayer)) {
               //do something
            }
}
```

File: README.md [hunk 3]
```diff
@@ -1,24 +1,256 @@
...
+```
+
+### Game room related functions
+All game related functionality is inside ```GameRoomController```. You must get a handle of this to invoke
```
Comment (KacperKluka) [inline README.md]: Shouldn't it be `All game room related functionality`? :wink:

File: README.md [hunk 4]
```diff
@@ -1,24 +1,256 @@
...
+```
+
+#### Query all players in a room
+You can query all players in a room like following. Notice that ```allPlayers()``` is a suspending function and needs
+ to be called from a coroutine scope.
+```
+someCoroutineScope.launch{
+   val allPlayers = allPlayers(room)
+
+}
```
Comment (KacperKluka) [inline README.md]: There's a needles blank line :wink: 
```suggestion
someCoroutineScope.launch{
   val allPlayers = allPlayers(room)
}
```

File: README.md [hunk 5]
```diff
@@ -1,24 +1,256 @@
...
+```
+
+
+
+
+
```
Comment (KacperKluka) [inline README.md]: Those blank lines can be removed :wink:

**Review by KacperKluka — CHANGES_REQUESTED** [review]
Looks good, I've just spotted a few small things :wink:

File: README.md (hunk 3 above)
Comment (ikbalkaya) [reply README.md]: Oops you are right

File: README.md (hunk 5 above)
Comment (ikbalkaya) [reply README.md:256]: https://github.com/ably-labs/GameRoomSDK/pull/8/commits/dd6af82937f959d3ec823a8824ebe449d91baec3

File: README.md (hunk 3 above)
Comment (ikbalkaya) [reply README.md]: https://github.com/ably-labs/GameRoomSDK/pull/8/commits/dd6af82937f959d3ec823a8824ebe449d91baec3

File: README.md (hunk 5 above)
Comment (ikbalkaya) [reply README.md]: https://github.com/ably-labs/GameRoomSDK/pull/8/commits/0f4b5c9a6d6f699f1936db0cc6638d096dac006e

**Comment by ikbalkaya** [comment]
@KacperKluka When applying the suggestions, Github automatically resolves conversation, so just so you know I didn't resolve them myself 😄

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Pull request merged**